idseq_dag/steps/generate_alignment_viz.py

```diff
@@ -399,6 +409,13 @@ def get_sequence_by_accession_id_s3(accession_id, entry, nt_bucket, nt_key):
                 # Get the sequence length based on the file size
                 seq_len = os.stat(accession_file).st_size
                 break
+            except IndexError as e:
+                # This may occur if the byterange fetched above is empty or does not properly align to an accession.
+                # This has occurred in the past when a reference cache issue caused the nt_loc_db and nt indices to be out of sync.
+                # Such issues should be investigated. However, the pipeline step can still complete with the missing data.
+                log.write("ntDbIndexError: Failed to get nt sequence by accession ID"
+                          f"{accession_id} {range_start} {range_end} {nt_bucket} {nt_key}: {e}")
```

Comment:
Per chat with Mark in person, need to move this to the code path that reads the range from local file.  Which is currently unprotected.

Reply:
Haha, actually I was wrong about this -- we are still going to be fetching these individually from s3.